### PR TITLE
Improve accessibility of registration form errors

### LINKED ISSUE #1394: Accessibility issues
A customer emailed these suggestions to us to fix some accessibility issues on the registration page.

> First item: 

> The list of form error messages is represented in <p> paragraph elements. For a11y, these should be semantically represented in a <ul> unordered list. Or, if the elements cannot be changed, use ARIA roles:

> `<div class="rcp_message" role="list">`
> `<p class="rcp_error" role="listitem">`

> Second item:

> Since the error messages are dynamically added to the page without refresh, the "rcp_message error" div needs to receive focus so that screen readers go straight to the error message and read it instead of waiting on the submit button with no indicators anything on page has changed.

> When errors are present, in addition to the error message above the submit, an error message should appear in the label of the input that contains an error.

> When submit button is pressed without filling out any information, no individual fields are in error but should be.

### PR DESCRIPTION
**Pull request by tnorthcutt**
Adds ARIA `list` and `listitem` roles to error message wrapper and individual messages, respectively.

Also focuses the error message div if it appears.

Closes #1394

Partially addresses #1621

### CODE REVIEW

File: includes/js/register.js [hunk 1]
```diff
@@ -101,6 +101,9 @@ jQuery(document).ready(function($) {
 
 		}).success(function( response ) {
 		}).done(function( response ) {
+            window.setTimeout(function() {
```
Comment (pippinsplugins) [inline includes/js/register.js]: Indention isn't quite right here.

File: includes/js/register.js [hunk 2]
```diff
@@ -101,8 +101,10 @@ jQuery(document).ready(function($) {
 
 		}).success(function( response ) {
 		}).done(function( response ) {
+			window.setTimeout(function() {
+				$('.rcp_message.error').attr("tabindex",-1).focus();
+			}, 10);
 		}).fail(function( response ) {
-			console.log( response );
```
Comment (nosegraze) [inline includes/js/register.js]: Why was this removed?

Comment (tnorthcutt) [reply includes/js/register.js]: Oops – I thought I'd introduced it in a previous commit. Do we actually want to leave a `console.log()` call running in production?

(FWIW older versions of IE don't support `console.log()`; not sure if that's a concern for us or not by now)

Comment (nosegraze) [reply includes/js/register.js]: I'd like to keep it due to the huge number of _"registration form not working / hanging"_ tickets we get in support. I have no idea if it's ever actually been useful, but we've had it there until now so I think we might as well keep it.

Comment (tnorthcutt) [reply includes/js/register.js]: Makes sense! Adding back.

Comment (tnorthcutt) [reply includes/js/register.js]: Got it!

**Comment by tnorthcutt** [comment]
I think this is ready to merge in for `2.9.11`
